Tools/Scripts/compare-static-analysis-results

```diff
@@ -24,77 +24,125 @@
-        help='path to scan-build'
+        help='Path to scan-build executable'
+    )
+    parser.add_argument(
+        '--check-expectations',
+        dest='check_expectations',
+        action='store_true',
+        default=False,
+        help='Compare new results to expectations (instead of a previous run)'
     )
     return parser.parse_args()
 
 
-def find_diff(file1, file2, mode):
-    # Create empty file if the corresponding one doesn't exist - this happens if a checker is added or removed
-    if not os.path.exists(f'{file1}-{mode}'):
-        f = open(f'{file1}-{mode}', 'a')
-        f.close()
-    if not os.path.exists(f'{file2}-{mode}'):
-        f = open(f'{file2}-{mode}', 'a')
-        f.close()
-
-    # Find new regressions
-    new_lines_list = []
-    find_issues_cmd = f"/usr/bin/grep -F -v -f {file1}-{mode} {file2}-{mode}"
-    try:
-        new_lines = subprocess.check_output(find_issues_cmd, shell=True, stderr=subprocess.STDOUT, text=True)
-        new_lines_list = new_lines.splitlines()
-    except subprocess.CalledProcessError as e:
-        if not e.returncode == 1:
-            sys.stderr.write(f'{e.output}')
-
-    # Find all fixes
-    fixed_lines_list = []
-    find_fixes_cmd = f'grep -F -v -f {file2}-{mode} {file1}-{mode}'
-    try:
-        fixed_lines = subprocess.check_output(find_fixes_cmd, shell=True, text=True, stderr=subprocess.STDOUT)
-        fixed_lines_list = fixed_lines.splitlines()
-    except subprocess.CalledProcessError as e:
-        if not e.returncode == 1:
-            sys.stderr.write(f'{e.output}')
-
-    return set(new_lines_list), set(fixed_lines_list)
-
-
-def compare_project_results(args, archive_path, new_path, project):
+def find_diff(args, file1, file2):
+    if not args.check_expectations:
+        # Create empty file if the corresponding one doesn't exist - this happens if a checker is added or removed
+        if not os.path.exists(file1):
+            f = open(f'file1', 'a')
+            f.close()
+        if not os.path.exists(f'file2'):
+            f = open(f'file2', 'a')
+            f.close()
+
+    with open(file1) as baseline_file, open(file2) as new_file:
+        baseline_list = baseline_file.read().splitlines()
+        new_file_list = new_file.read().splitlines()
+        # Find new regressions
+        diff_new_from_baseline = set(new_file_list) - set(baseline_list)
+         # Find fixes
+        diff_baseline_from_new = set(baseline_list) - set(new_file_list)
+
+    return set(diff_new_from_baseline), set(diff_baseline_from_new)
+
+
+def create_filtered_results_dir(args, project, issues, category='StaticAnalyzerRegressions'):
+    # Create symlinks to new issues only so that we can run scan-build to generate new index.html files
+    path_to_reports = os.path.abspath(f'{args.build_output}/{category}/{project}/StaticAnalyzerReports')
+    subprocess.run(['mkdir', '-p', path_to_reports])
+    for issue_hash in issues:
+        report = f"report-{issue_hash[:6]}.html"
```

Comment:
Shouldn't this be 12 characters? Why only 6?

Reply:
`scan-build` uses the first 6 characters of an issue hash to name the report, like `report-7ab18e.html`